# Lista de exercícios #2 - Programação e Análise de Dados com Python

## Pedro Milreu Cunha - 20201003111 - pedro.milreu@academico.ufpb.br

## Ciclos *while*

**1- Escreva um programa que solicite que o usuário digite um número inteiro
   par e positivo. No final, após ele ter informado vários número (um por vez)
   o resultado deve ser a soma de todos os valores.**


In [6]:
entrada = 0
soma = 0 

while True:
    entrada = int(input("Digite um número inteiro par e positivo."))
    print(f"Número digitado:{entrada}.")
    if (entrada < 0 or entrada % 2 != 0):
        print(f"Encerrando o programa. A soma dos números inteiros pares e positivos digitados é {soma}.")
        break
    else:
        soma += entrada



Número digitado:4.
Número digitado:6.
Número digitado:8.
Número digitado:10.
Número digitado:12.
Número digitado:44.
Número digitado:102.
Número digitado:982.
Número digitado:5.
Encerrando o programa. A soma dos números inteiros pares e positivos digitados é 1168.


**2- Escreva um programa de jogo de adivinhação. Solicite
   que o usuário digite um número entre 0 e 100. Na primeira tentativa, se não
   for a correta, imprima uma mensagem afirmando que seu palpite está "quente".
   Para todas as demais tentativas, imprima "quente" se o novo número informado
   for estritamente próximo do número secreto e "frio" caso contrário. Continue
   solicitando para ele informar um novo número até que o número secreto seja
   informado.** 


In [12]:
import random # Módulo para gerar número pseudo-aleatórios 
num_secreto = random.randint(0, 100) # Embora o enunciado não informe que os números devam ser inteiros, optei por essa
# opção para que o número total de possibilidades fosse finito.
dist_max = 5

entrada = int(input("Digite um número inteiro entre 0 e 100."))

if (entrada == num_secreto):
    print(f"Parabéns! Você acertou. O número secreto era {num_secreto}.")
else:
    print(f"Seu palpite, {entrada}, está quente. Continue tentando!")
    while(entrada != num_secreto):
        entrada = int(input("Tente novamente."))
        if (abs(entrada - num_secreto) <= dist_max and (entrada - num_secreto) != 0):
            print(f"Seu palpite, {entrada}, está quente. Continue tentando!")
        if (abs(entrada - num_secreto) > dist_max):
            print(f"Seu palpite, {entrada}, está frio. Não desista!")
    print(f"Parabéns! Você acertou. O número secreto era {num_secreto}.")        

    



Seu palpite, 10, está quente. Continue tentando!
Seu palpite, 50, está frio. Não desista!
Seu palpite, 40, está frio. Não desista!
Seu palpite, 30, está frio. Não desista!
Seu palpite, 80, está frio. Não desista!
Seu palpite, 70, está quente. Continue tentando!
Seu palpite, 65, está frio. Não desista!
Seu palpite, 72, está quente. Continue tentando!
Seu palpite, 73, está quente. Continue tentando!
Parabéns! Você acertou. O número secreto era 74.


**3- Escreva um programa que com valores inicias para as seguintes variáveis:
   a, b, c = 0, 0, 0 . Como resultado, imprima todas as permutações possíveis
   para essas três variáveis. Exemplo do resultado esperado:**
   ```
   Output : 000 , 001 ,002, 003, 004, 005 ,006, 007, 008, 009, 010, 011 …… 999
   ```


In [24]:
a = int(input("Digite um valor inicial para a variável 'a'."))
b = int(input("Digite um valor inicial para a variável 'b'."))
c = int(input("Digite um valor inicial para a variável 'c'."))
print(f"Parâmetros iniciais: (a,b,c) = ({a},{b},{c}). Exibindo as permutações:\n")

centenas = 0
while (centenas <= a):
    dezenas = 0
    while(dezenas <= b):
        unidades = 0
        while(unidades <= c):
            if (centenas == a and dezenas == b and unidades == c):
                print(f"{centenas}{dezenas}{unidades}")
            else:
                print(f"{centenas}{dezenas}{unidades}", end = ",") # O parâmetro "end" permite controlar o comportamento de fim-de-linha
            unidades += 1
        dezenas += 1
    centenas += 1




Parâmetros iniciais: (a,b,c) = (9,6,8). Exibindo as permutações:

000,001,002,003,004,005,006,007,008,010,011,012,013,014,015,016,017,018,020,021,022,023,024,025,026,027,028,030,031,032,033,034,035,036,037,038,040,041,042,043,044,045,046,047,048,050,051,052,053,054,055,056,057,058,060,061,062,063,064,065,066,067,068,100,101,102,103,104,105,106,107,108,110,111,112,113,114,115,116,117,118,120,121,122,123,124,125,126,127,128,130,131,132,133,134,135,136,137,138,140,141,142,143,144,145,146,147,148,150,151,152,153,154,155,156,157,158,160,161,162,163,164,165,166,167,168,200,201,202,203,204,205,206,207,208,210,211,212,213,214,215,216,217,218,220,221,222,223,224,225,226,227,228,230,231,232,233,234,235,236,237,238,240,241,242,243,244,245,246,247,248,250,251,252,253,254,255,256,257,258,260,261,262,263,264,265,266,267,268,300,301,302,303,304,305,306,307,308,310,311,312,313,314,315,316,317,318,320,321,322,323,324,325,326,327,328,330,331,332,333,334,335,336,337,338,340,341,342,343,344,345,346,347,34

**4- Elabore um programa que imprima todos os números primos de
   entre 0 to 100.**

In [26]:
contador = 2

while (contador <= 100):
    if (contador == 2):
        print(contador)
        contador += 1
    elif (contador % 2 == 0):
        contador += 1
    else:
        temp = 2
        divisores = 1
        while (divisores == 1 and temp < contador):
            if (contador % temp == 0):
                divisores += 1
            else:
                temp += 1
        if (divisores == 1):
            print(contador)
        contador += 1


2
3
5
7
11
13
17
19
23
29
31
37
41
43
47
53
59
61
67
71
73
79
83
89
97


## *Listas*

**1- Faça um Programa que leia 4 notas, mostre as notas e a média na tela.**
 

In [32]:
notas = []
contador = 0
soma = 0

while (contador <= 3):
    notas.append(float(input(f"Digite a {contador+1}ª nota:")))
    soma += notas[contador]
    contador += 1
media = soma/(contador) 

print(f"As notas digitadas foram {notas} e a média é {media:.2f}.")


As notas digitadas foram [5.0, 6.0, 9.0, 10.0] e a média é 7.50.


**2- Faça um Programa que leia 20 números inteiros e armazene-os num vetor. Armazene os números pares no vetor PAR e os
    números IMPARES no vetor impar. Imprima os três vetores.**

In [38]:
numeros = []
par = []
impar = []

contador = 0

while (contador <= 19):
    numeros.append(int(input(f"Digite o {contador+1}º número:.")))
    if (numeros[contador] % 2 == 0):
        par.append(numeros[contador])
    else:
        impar.append(numeros[contador])
    contador += 1

print(f"Vetor 'números': {numeros}.")
print(f"Vetor 'par': {par}.")
print(f"Vetor 'ímpar': {impar}")

Vetor 'números': [4, 12, 13, 16, 153, 4620, 5521, 741, 130, 22, 103, 201, 254, 166, 746, 302, 134, 222, 201, 103].
Vetor 'par': [4, 12, 16, 4620, 130, 22, 254, 166, 746, 302, 134, 222].
Vetor 'ímpar': [13, 153, 5521, 741, 103, 201, 201, 103]


**3- Faça um Programa que peça as quatro notas de 10 alunos, calcule e armazene num vetor a média de cada aluno, imprima
   o número de alunos com média maior ou igual a 7.0.**


In [45]:
num_alunos = 10
num_notas = 4
medias = []
num_acima_7 = 0

contador_alunos = 0

while (contador_alunos < num_alunos):
    contador_notas = 0
    notas = []
    soma = 0
    while(contador_notas < num_notas):
        notas.append(float(input(f"Digite a {contador_notas+1}ª nota:")))
        soma += notas[-1]
        contador_notas += 1
    medias.append(soma/contador_notas)
    if (medias[-1] >= 7):
        num_acima_7 += 1
    contador_alunos += 1

print (f"O vetor de médias é {medias} e, portanto, o número de alunos com média maior ou igual a 7 foi de: {num_acima_7}.")

O vetor de médias é [8.25, 6.25, 6.0, 7.5, 5.25, 10.0, 7.5, 6.0, 7.25, 7.25] e, portanto, o número de alunos com média maior ou igual a 7 foi de: 6.


**4- Elabore um programa que receba a temperatura média de cada mês do ano e armazene-as em uma lista.
  Após isto, calcule a média anual das temperaturas e mostre todas as temperaturas acima da média anual, e em que mês
  elas ocorreram (mostrar o mês por extenso: 1 – Janeiro, 2 – Fevereiro, . . . ).**

In [48]:
meses = ["Janeiro", "Fevereiro", "Março", "Abril", "Maio", "Junho", "Julho", "Agosto", "Setembro", "Outubro", "Novembro", "Dezembro"]
temp_medias = []
contador = 0
soma = 0
while (contador < len(meses)):
    temp_medias.append(float(input(f"Digite a temperatura média do mês de {meses[contador]}:")))
    soma += temp_medias[-1]
    contador += 1

media_anual = soma/contador
contador = 0 
while (contador < len(temp_medias)):
    if (temp_medias[contador] > media_anual):
        print(f"Temperatura de {temp_medias[contador]}ºC (média anual de {media_anual:.2f}ºC) no mês de {meses[contador]}.")
    contador += 1


Temperatura de 30.0ºC (média anual de 28.76ºC) no mês de Janeiro.
Temperatura de 32.4ºC (média anual de 28.76ºC) no mês de Fevereiro.
Temperatura de 30.0ºC (média anual de 28.76ºC) no mês de Março.
Temperatura de 32.6ºC (média anual de 28.76ºC) no mês de Junho.
Temperatura de 33.0ºC (média anual de 28.76ºC) no mês de Julho.
Temperatura de 32.0ºC (média anual de 28.76ºC) no mês de Outubro.
Temperatura de 33.0ºC (média anual de 28.76ºC) no mês de Novembro.
Temperatura de 29.6ºC (média anual de 28.76ºC) no mês de Dezembro.


**5- Utilizando listas faça um programa que faça 5 perguntas para uma pessoa sobre um crime. As perguntas são:**
   
   "Telefonou para a vítima?"
   "Esteve no local do crime?"
   "Mora perto da vítima?"
   "Devia para a vítima?"
   "Já trabalhou com a vítima?"

   **O programa deve no final emitir uma classificação sobre a participação da pessoa no crime. Se a pessoa responder
   positivamente a 2 questões ela deve ser classificada como "Suspeita", entre 3 e 4 como "Cúmplice" e
    5 como "Assassino". Caso contrário, ele será classificado como "Inocente".**

In [50]:
perguntas = ["Telefonou para a vítima?", "Esteve no local do crime?", "Mora perto da vítima?", "Devia para vítima?",
"Já trabalhou com a vítima?"]
soma = 0
contador = 0

while (contador < len(perguntas)):
    resposta = input(f"{perguntas[contador]} (S/N)").lower()
    if (resposta == "s"):
        soma += 1
    contador += 1

if (soma == 2):
    print("Classificação: Suspeito(a)")
elif (soma == 3 or soma == 4):
    print("Classificação: Cúmplice")
elif (soma == 5):
    print("Classificação: Assassino(a)")
else:
    print("Classificação: Inocente")

Classificação: Cúmplice


**6- Faça um programa que leia um número indeterminado de valores, correspondentes a notas, encerrando a entrada de dados
   quando for informado um valor igual a -1 (que não deve ser armazenado). Após esta entrada de dados, faça:**

   - Mostre a quantidade de valores que foram lidos;
   - Exiba todos os valores na ordem em que foram informados, um ao lado do outro;
   - Exiba todos os valores na ordem inversa à que foram informados, um abaixo do outro;
   - Calcule e mostre a soma dos valores;
   - Calcule e mostre a média dos valores;
   - Calcule e mostre a quantidade de valores acima da média calculada;
   - Calcule e mostre a quantidade de valores abaixo de sete;
   - Encerre o programa com uma mensagem;

In [57]:
notas = []
soma = 0
num_acima_media = 0
num_abaixo_7 = 0
contador = 0

while True:
    entrada = float(input(f"Digite a {contador+1}ª nota: (-1 para encerrar o programa)"))
    if (entrada == -1):
        break
    else:
        notas.append(entrada)
        soma += notas[-1]
        contador += 1

media = soma/contador

contador = 0
while (contador < len(notas)):
    if (notas[contador] > media):
        num_acima_media += 1
    if (notas[contador] < 7):
        num_abaixo_7 += 1
    contador += 1

print(f"Foram lidas {len(notas)} notas.")

print("\nNotas na ordem em que foram inseridos uma ao lado da outra:")
contador = 0 
while (contador < len(notas)):
    print(f"{notas[contador]}", end = " ")
    contador += 1

print("\n\nNotas na ordem inversa à que foram inseridos uma abaixo da outra:")
contador = -1
while (contador >= -len(notas)):
    print(f"{notas[contador]}")
    contador -= 1

print(f"\nA soma das notas inseridas é: {soma:.2f}.")
print(f"\nA média das notas inseridas é: {media:.2f}.")
print(f"\nO número de notas acima da média de {media:.2f} é: {num_acima_media}.")
print(f"\nO número de notas abaixo de 7 é: {num_abaixo_7}.")
print("\n --------------------------------------------------- \n \n \t\t\t Encerrando o programa. Obrigado por utilizá-lo!")

Foram lidas 12 notas.

Notas na ordem em que foram inseridos uma ao lado da outra:
9.6 9.25 7.0 6.0 4.5 10.0 9.0 9.6 7.0 10.0 9.8 10.0 

Notas na ordem inversa à que foram inseridos uma abaixo da outra:
10.0
9.8
10.0
7.0
9.6
9.0
10.0
4.5
6.0
7.0
9.25
9.6

A soma das notas inseridas é: 101.75.

A média das notas inseridas é: 8.48.

O número de notas acima da média de 8.48 é: 8.

O número de notas abaixo de 7 é: 2.

 --------------------------------------------------- 
 
 			 Encerrando o programa. Obrigado por utilizá-lo!


**7- Utilize uma lista para resolver o problema a seguir. Uma empresa paga seus vendedores com base em comissões.
   O vendedor recebe 200 reais por semana mais 9 por cento de suas vendas brutas daquela semana. Por exemplo, um vendedor
   que teve vendas brutas de 3000 reais em uma semana recebe 200 reais mais 9 por cento de 3000 reais, ou seja,
   um total de 470 reais. Escreva um programa (usando um lista de contadores) que determine quantos vendedores
   receberam salários nos seguintes intervalos de valores:**
    - <span>R$</span>200 - R$299
    - <span>R$</span>300 - R$399
    - <span>R$</span>400 - R$499
    - <span>R$</span>500 - R$599
    - <span>R$</span>600 - R$699
    - <span>R$</span>700 - R$799
    - <span>R$</span>800 - R$899
    - <span>R$</span>900 - R$999
    - <span>R$</span>1.000 em diante

In [64]:
intervalos_salarios = [["R$200 - R$299", 0], ["R$300 - R$399", 0], ["R$400 - R$499", 0], ["R$500 - R$599", 0], ["R$600 - R$699", 0],
["R$700 - R$799", 0], ["R$800 - R$899", 0], ["R$900 - R$999", 0], ["R$1.000+", 0]]
contador = 1

while True:
    entrada = float(input(f"Digite o valor das vendas brutas da semana do vendedor {contador} (-1 para encerrar):"))
    if (entrada == -1):
        break
    else:
        salario = 200 + 0.09*entrada
        if (200 <= salario <= 299):
            intervalos_salarios[0][1] += 1
        elif (300 <= salario <= 399):
            intervalos_salarios[1][1] += 1
        elif (400 <= salario <= 499):
            intervalos_salarios[2][1] += 1
        elif (500 <= salario <= 599):
            intervalos_salarios[3][1] += 1
        elif (600 <= salario <= 699):
            intervalos_salarios[4][1] += 1
        elif (700 <= salario <= 799):
            intervalos_salarios[5][1] += 1
        elif (800 <= salario <= 899):
            intervalos_salarios[6][1] += 1
        elif (900 <= salario <= 999):
            intervalos_salarios[7][1] += 1
        else:
            intervalos_salarios[8][1] += 1
    contador += 1

print("Situação dos salários dos vendedores: \n --------------------------------------------------")
contador = 0
while (contador < len(intervalos_salarios)):
    if (contador == len(intervalos_salarios) - 1):
        print(f"Intervalo: {intervalos_salarios[contador][0]} \t\t Nº de vendedores: {intervalos_salarios[contador][1]} \n \
        --------------------------------------------------")
    else:        
        print(f"Intervalo: {intervalos_salarios[contador][0]} \t Nº de vendedores: {intervalos_salarios[contador][1]}")
    contador += 1

Situação dos salários dos vendedores: 
 --------------------------------------------------
Intervalo: R$200 - R$299 	 Nº de vendedores: 0
Intervalo: R$300 - R$399 	 Nº de vendedores: 2
Intervalo: R$400 - R$499 	 Nº de vendedores: 3
Intervalo: R$500 - R$599 	 Nº de vendedores: 0
Intervalo: R$600 - R$699 	 Nº de vendedores: 1
Intervalo: R$700 - R$799 	 Nº de vendedores: 2
Intervalo: R$800 - R$899 	 Nº de vendedores: 2
Intervalo: R$900 - R$999 	 Nº de vendedores: 0
Intervalo: R$1.000+ 		 Nº de vendedores: 1 
 --------------------------------------------------


**8- Uma grande emissora de televisão quer fazer uma enquete entre os seus telespectadores para saber qual o melhor
   jogador após cada jogo. Para isto, faz-se necessário o desenvolvimento de um programa, que será utilizado pelas
    telefonistas, para a computação dos votos. Sua equipe foi contratada para desenvolver este programa, utilizando
    a linguagem de programação Python. Para computar cada voto, a telefonista digitará um número, entre 1 e 23,
    correspondente ao número da camisa do jogador. Um número de jogador igual zero, indica que a votação foi encerrada.
    Se um número inválido for digitado, o programa deve ignorá-lo, mostrando uma breve mensagem de aviso, e voltando a
    pedir outro número. Após o final da votação, o programa deverá exibir**:

   - O total de votos computados;
   - Os númeos e respectivos votos de todos os jogadores que receberam votos;
   - O percentual de votos de cada um destes jogadores;
   - O número do jogador escolhido como o melhor jogador da partida, juntamente com o número de votos e o percentual de votos dados a ele.
   
**Observe que os votos inválidos e o zero final não devem ser computados como votos. O resultado aparece ordenado
pelo número do jogador. O programa deve fazer uso de arrays. O programa deverá executar o cálculo do percentual
de cada jogador através de uma função. Esta função receberá dois parâmetros: o número de votos de um jogador e o
total de votos. A função calculará o percentual e retornará o valor calculado.
Abaixo segue uma tela de exemplo. O disposição das informações deve ser o mais próxima possível ao exemplo.
Os dados são fictícios e podem mudar a cada execução do programa.**

   ```
   Enquete: Quem foi o melhor jogador?

   Número do jogador (0=fim): 9
   Número do jogador (0=fim): 10
   Número do jogador (0=fim): 9
   Número do jogador (0=fim): 10
   Número do jogador (0=fim): 11
   Número do jogador (0=fim): 10
   Número do jogador (0=fim): 50
   Informe um valor entre 1 e 23 ou 0 para sair!
   Número do jogador (0=fim): 9
   Número do jogador (0=fim): 9
   Número do jogador (0=fim): 0

   Resultado da votação:

   Foram computados 8 votos.

   Jogador Votos           %
   9               4               50,0%
   10              3               37,5%
   11              1               12,5%
   O melhor jogador foi o número 9, com 4 votos, correspondendo a 50% do total de votos.
```


In [72]:
lista_camisas = [*range(1, 24)] # O "*" serve para 'desmontar' o range(1, 23) em uma sequência de valores
lista_votos = [0]*23
votos_computados = 0 

print("Enquete: Quem foi o melhor jogador?")

entrada = -1
while (entrada != 0):
    entrada = int(input("Número do jogador (0 = fim):"))
    if (entrada in lista_camisas):
        lista_votos[entrada-1] += 1
        votos_computados += 1
    else:
        print("Informe um valor entre 1 e 23 ou 0 para sair.")

contador = 0
percentual = []
while (contador < len(lista_votos)):
    percentual.append(lista_votos[contador]/votos_computados)
    contador += 1

print("Resultado da votação: \n")
print(f"Foram computados {votos_computados} votos.")
print("Jogador \t Votos \t %")
contador = 0
camisa_max = 0
while (contador < len(lista_votos)):
    if (lista_votos[contador] != 0):
        print(f"{lista_camisas[contador]} \t\t {lista_votos[contador]} \t {100*percentual[contador]:.2f}%")
    if (lista_votos[contador] == max(lista_votos)):
        camisa_max = contador + 1
    contador += 1
print(f"O melhor jogador foi o número {camisa_max}, com {lista_votos[camisa_max-1]} votos,\
correspondendo a {100*percentual[camisa_max-1]:.2f}% dos votos.")

# no caso de empate de votos o jogador com a camisa de menor número é escolhido como melhor




Enquete: Quem foi o melhor jogador?
Informe um valor entre 1 e 23 ou 0 para sair.
Informe um valor entre 1 e 23 ou 0 para sair.
Informe um valor entre 1 e 23 ou 0 para sair.
Resultado da votação: 

Foram computados 10 votos.
Jogador 	 Votos 	 Percentual
1 		 3 		 30.00%
2 		 1 		 10.00%
3 		 1 		 10.00%
11 		 5 		 50.00%
O melhor jogador foi o número 11, com 5 votos,correspondendo a 50.00% dos votos.


**9- Uma empresa de pesquisas precisa tabular os resultados da seguinte enquete feita a um grande quantidade de
  organizações:**

```
"Qual o melhor Sistema Operacional para uso em servidores?"

As possíveis respostas são:

1- Windows Server
2- Unix
3- Linux
4- Netware
5- Mac OS
6- Outro
```
**Você foi contratado para desenvolver um programa que leia o resultado da enquete e informe ao final o resultado da mesma.
O programa deverá ler os valores até ser informado o valor 0, que encerra a entrada dos dados. Não deverão ser aceitos
valores além dos válidos para o programa (0 a 6). Os valores referentes a cada uma das opções devem ser armazenados
num vetor. Após os dados terem sido completamente informados, o programa deverá calcular a percentual de cada um dos
concorrentes e informar o vencedor da enquete. O formato da saída foi dado pela empresa, e é o seguinte:**

```
Sistema Operacional     Votos   %
-------------------     -----   ---
Windows Server           1500   17%
Unix                     3500   40%
Linux                    3000   34%
Netware                   500    5%
Mac OS                    150    2%
Outro                     150    2%
-------------------     -----
Total                    8800

O Sistema Operacional mais votado foi o Unix, com 3500 votos, correspondendo a 40% dos votos.

In [80]:
lista_sistemas = ["Windows Server", "Unix", "Linux", "Netware", "Mac OS", "Outro"]
lista_votos = [0]*6
votos_computados = 0 

entrada = -1
while (entrada != 0):
    entrada = int(input("Voto:"))
    if (entrada in range(1, 7)):
        lista_votos[entrada-1] += 1
        votos_computados += 1
    else:
        print("Entrada inválida")

contador = 0
percentual = []
while (contador < len(lista_votos)):
    percentual.append(lista_votos[contador]/votos_computados)
    contador += 1

print("Sistema Operacional \t Votos \t %")
print("------------------- \t ----- \t ---")
contador = 0
camisa_max = 0
while (contador < len(lista_votos)):
    print('{:19s} \t {:5} \t {:6.2f}%'.format(lista_sistemas[contador], lista_votos[contador], 100*percentual[contador]))
    if (lista_votos[contador] == max(lista_votos)):
        sistema_max = contador
    contador += 1
print("------------------- \t -----")
print('{:19s} \t {:5}'.format("Total", votos_computados))
print(f"O Sistema Operacional mais votado foi o {lista_sistemas[sistema_max]}, com {lista_votos[sistema_max]} votos,\
correspondendo a {100*percentual[sistema_max]:.2f}% dos votos.")


Entrada inválida
Sistema Operacional 	 Votos 	 %
------------------- 	 ----- 	 ---
Windows Server      	     6 	  54.55%
Unix                	     0 	   0.00%
Linux               	     0 	   0.00%
Netware             	     0 	   0.00%
Mac OS              	     5 	  45.45%
Outro               	     0 	   0.00%
------------------- 	 -----
Total               	    11
O Sistema Operacional mais votado foi o Windows Server, com 6 votos,correspondendo a 54.55% dos votos.


**10- As Organizações Tabajara resolveram dar um abono aos seus colaboradores em reconhecimento ao bom resultado alcançado
durante o ano que passou. Para isto contratou você para desenvolver a aplicação que servirá como uma projeção de quanto
será gasto com o pagamento deste abono. Após reuniões envolvendo a diretoria executiva, a diretoria financeira e os representantes do sindicato laboral, chegou-se a seguinte forma de cálculo:**

    - Cada funcionário receberá o equivalente a 20% do seu salário bruto de dezembro;
    - O piso do abono será de 100 reais, isto é, aqueles funcionários cujo salário for muito baixo, recebem este
      valor mínimo;

**Neste momento, não se deve ter nenhuma preocupação com colaboradores com tempo menor de casa, descontos, impostos ou
outras particularidades.**

**Seu programa deverá permitir a digitação do salário de um número indefinido (desconhecido) de salários. Um valor de salário igual a 0 (zero) encerra a digitação. Após a entrada de todos os dados o programa deverá calcular o valor do abono concedido a cada colaborador, de acordo com a regra definida acima. Ao final, o programa deverá apresentar:**

- O salário de cada funcionário, juntamente com o valor do abono;
- O número total de funcionário processados;
- O valor total a ser gasto com o pagamento do abono;
- O número de funcionário que receberá o valor mínimo de 100 reais;
- O maior valor pago como abono; A tela abaixo é um exemplo de execução do programa, apenas para fins ilustrativos. Os valores podem mudar a cada execução do programa.

```
Projeção de Gastos com Abono
============================

Salário: 1000
Salário: 300
Salário: 500
Salário: 100
Salário: 4500
Salário: 0

Salário    - Abono
R$ 1000.00 - R$  200.00
R$  300.00 - R$  100.00
R$  500.00 - R$  100.00
R$  100.00 - R$  100.00
R$ 4500.00 - R$  900.00

Foram processados 5 colaboradores
Total gasto com abonos: R$ 1400.00
Valor mínimo pago a 3 colaboradores
Maior valor de abono pago: R$ 900.00
```

In [81]:
salarios = []
abonos = []
num_abonos_min = 0
num_colaboradores = 0
entrada = -1

while (entrada != 0):
    entrada = float(input("Salário:"))
    if (entrada > 0):
        num_colaboradores += 1
        salarios.append(entrada)
        abonos.append(max(100, 0.2*entrada))
        if (abonos[-1] == 100):
            num_abonos_min += 1
    elif(entrada != 0):
        print("Valor inválido.")

total_gasto_abonos = sum(abonos)
max_abono = max(abonos)

print("Projeção de gastos com abono: \n============================")
print("Salário \t - \t Abono")
contador = 0
while (contador < len(salarios)):
    print('R${:10.2f} \t {:5s} \t R${:10.2f}'.format(salarios[contador], "-", abonos[contador]))
    contador += 1

print(f"Foram processados {num_colaboradores} colaboradores.")
print(f"Total gasto com abonos: R${total_gasto_abonos:.2f}")
print(f"Valor mínimo pago a {num_abonos_min} colaboradores.")
print(f"Maior valor de abono pago: R${max_abono:.2f}")
    

Valor inválido.
Projeção de gastos com abono: 
Salário 	 - 	 Abono
R$    600.00 	 -     	 R$    120.00
R$    562.65 	 -     	 R$    112.53
R$   6000.00 	 -     	 R$   1200.00
R$  12000.00 	 -     	 R$   2400.00
R$    230.65 	 -     	 R$    100.00
R$   1000.00 	 -     	 R$    200.00
Foram processados 6 colaboradores.
Total gasto com abonos: R$4132.53
Valor mínimo pago a 1 colaboradores.
Maior valor de abono pago: R$2400.00


**11. Faça um programa que carregue uma lista com os modelos de cinco carros (exemplo de modelos: FUSCA, GOL, VECTRA etc).
    Carregue uma outra lista com o consumo desses carros, isto é, quantos quilômetros cada um desses carros faz com um
    litro de combustível. Calcule e mostre:**

    - O modelo do carro mais econômico;
    - Quantos litros de combustível cada um dos carros cadastrados consome para percorrer uma distância de 1000 quilômetros e quanto isto custará, considerando um que a gasolina custe 2,25 reais o litro.

**Abaixo segue uma tela de exemplo. O disposição das informações deve ser o mais próxima possível ao exemplo.
Os dados são fictícios e podem mudar a cada execução do programa.**

```
Comparativo de Consumo de Combustível

Veículo 1
Nome: fusca
Km por litro: 7
Veículo 2
Nome: gol
Km por litro: 10
Veículo 3
Nome: uno
Km por litro: 12.5
Veículo 4
Nome: Vectra
Km por litro: 9
Veículo 5
Nome: Peugeout
Km por litro: 14.5

Relatório Final
 1 - fusca           -    7.0 -  142.9 litros - R$ 321.43
 2 - gol             -   10.0 -  100.0 litros - R$ 225.00
 3 - uno             -   12.5 -   80.0 litros - R$ 180.00
 4 - vectra          -    9.0 -  111.1 litros - R$ 250.00
 5 - peugeout        -   14.5 -   69.0 litros - R$ 155.17
O menor consumo é do peugeout.
```


**

In [89]:
lista_modelos = []
lista_km_litro = []
preco_litro_gasolina = 2.25
dist = 1000

print("Comparativo de Consumo de Combustível")

contador = 0 
while(contador < 5):
    print(f"Veículo {contador+1}")
    lista_modelos.append(input("Nome:"))
    lista_km_litro.append(float(input("Km por litro:")))
    contador += 1

print("Relatório Final")
contador = 0
indice_max = 0
while (contador < 5):
    print('{:3d} \t {:3s} \t {:15s} \t {:3s} \t {:6.1f} \t {:3s} \t {:6.1f} litros \t {:3s} \t R${:6.2f}'.format(contador+1, "-", lista_modelos[contador],
    "-", lista_km_litro[contador], "-", dist/lista_km_litro[contador], "-", preco_litro_gasolina*dist/lista_km_litro[contador]))
    if (lista_km_litro[contador] == max(lista_km_litro)):
        indice_max = contador
    contador += 1
print(f"O menor consumo é do {lista_modelos[indice_max]}.")

Comparativo de Consumo de Combustível
Veículo 1
Veículo 2
Veículo 3
Veículo 4
Veículo 5
Relatório Final
  1 	 -   	 Fusca           	 -   	    7.0 	 -   	  142.9 litros 	 -   	 R$321.43
  2 	 -   	 Gol             	 -   	   10.0 	 -   	  100.0 litros 	 -   	 R$225.00
  3 	 -   	 Uno             	 -   	   12.5 	 -   	   80.0 litros 	 -   	 R$180.00
  4 	 -   	 Vectra          	 -   	    9.0 	 -   	  111.1 litros 	 -   	 R$250.00
  5 	 -   	 Peugeout        	 -   	   14.5 	 -   	   69.0 litros 	 -   	 R$155.17
O menor consumo é do Peugeout.
